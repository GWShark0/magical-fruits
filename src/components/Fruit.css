.fruit {
  margin: 10px;
  cursor: pointer;
}

.fruit__preview {
  width: 200px;
  height: 200px;
  border: 6px solid transparent;
  border-radius: 3px;
  transition: border-color 150ms;
}

.fruit--selected .fruit__preview {
  border-color: #34A853;
}
