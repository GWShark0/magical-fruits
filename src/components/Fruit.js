import React from 'react';
import classNames from 'classnames';
import fruits from '../fruits';

import './Fruit.css';

function Fruit({ name, selected, onClick }) {

  const className = classNames(
    'fruit',
    { 'fruit--selected': selected },
  );

  return (
    <div className={className} onClick={() => onClick(name)}>
      <img className="fruit__preview" src={fruits[name]} alt={name} />
    </div>
  )
}

export default Fruit;
