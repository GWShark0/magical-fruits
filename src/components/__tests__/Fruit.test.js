import React from 'react';
import renderer from 'react-test-renderer';
import Fruit from '../Fruit';

it('should render', () => {
  const component = renderer.create(
    <Fruit name="apple" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it('should toggle selected state', () => {
  const component = renderer.create(
    <Fruit name="banana" />
  );
  let tree;

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onClick();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
