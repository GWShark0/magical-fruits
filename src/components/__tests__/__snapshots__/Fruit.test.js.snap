exports[`test should render 1`] = `
<div
  className="fruit"
  onClick={[Function]}>
  <img
    alt="apple"
    className="fruit__preview"
    src="apple.jpg" />
</div>
`;
